### ISSUE TITLE
Action couldn't run with specifying argument which includes multi-byte characters

### ISSUE THREAD

**Issue by userlocalhost** [StackStorm/st2#3107]
This is a step to reproduce.
```
$ st2 run core.local cmd='echo ｈｏｇｅ'
.
id: 584f6df66f086f5adb3c499b
status: failed
parameters: 
  cmd: "echo ｈｏｇｅ"
result: 
  error: '''ascii'' codec can''t encode characters in position 39-42: ordinal not in range(128)'
  traceback: "  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/st2actions/container/base.py", line 99, in _do_run
    (status, result, context) = runner.run(action_params)
  File "/opt/stackstorm/runners/local_runner/local_runner.py", line 152, in run
    kill_func=kill_process)
  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/st2common/util/green/shell.py", line 87, in run_command
    env=env, cwd=cwd, shell=shell, preexec_fn=preexec_func)
  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/eventlet/green/subprocess.py", line 54, in __init__
    subprocess_orig.Popen.__init__(self, args, 0, *argss, **kwds)
  File "/usr/lib/python2.7/subprocess.py", line 710, in __init__
    errread, errwrite)
  File "/usr/lib/python2.7/subprocess.py", line 1327, in _execute_child
    raise child_exception
"
```

**Comment by humblearner** [comment]
Hello @userlocalhost, can you share the `st2 --version` and OS ?
As it is working for me on latest release, centos72: 

```
[vagrant@centos72 ~]$ st2 run core.local cmd='echo ｈｏｇｅ'
.
id: 584f75804c3a5307cfa4e730
status: succeeded
parameters:
  cmd: "echo ｈｏｇｅ"
result:
  failed: false
  return_code: 0
  stderr: ''
  stdout: "ｈｏｇｅ"
  succeeded: true
[vagrant@centos72 ~]$ st2 run core.local cmd='echo ｈｏｇｅ' -j

{
    "id": "584f75864c3a5307cfa4e733",
    "parameters": {
        "cmd": "echo \uff48\uff4f\uff47\uff45"
    },
    "result": {
        "failed": false,
        "return_code": 0,
        "stderr": "",
        "stdout": "\uff48\uff4f\uff47\uff45",
        "succeeded": true
    },
    "status": "succeeded"
}

[vagrant@centos72 ~]$ st2 --version
st2 2.1.0
```

**Comment by userlocalhost** [comment]
Hi, @humblearner 
Thank you for your reply. Here is my environment.
```
vagrant@st2-node:~$ st2 --version
st2 2.1.0
vagrant@st2-node:~$ dpkg -l st2
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name              Version       Architecture  Description
+++-=================-=============-=============-=======================================
ii  st2               2.1.0-16      amd64         StackStorm Event-driven automation
vagrant@st2-node:~$ cat /etc/lsb-release 
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=14.04
DISTRIB_CODENAME=trusty
DISTRIB_DESCRIPTION="Ubuntu 14.04.5 LTS"
vagrant@st2-node:~$ st2 run core.local cmd='echo ｈｏｇｅ'
.
id: 584f7d5e6f086f5adb3c49a4
status: failed
parameters: 
  cmd: "echo ｈｏｇｅ"
result: 
  error: '''ascii'' codec can''t encode characters in position 39-42: ordinal not in range(128)'
  traceback: "  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/st2actions/container/base.py", line 99, in _do_run
    (status, result, context) = runner.run(action_params)
  File "/opt/stackstorm/runners/local_runner/local_runner.py", line 152, in run
    kill_func=kill_process)
  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/st2common/util/green/shell.py", line 87, in run_command
    env=env, cwd=cwd, shell=shell, preexec_fn=preexec_func)
  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/eventlet/green/subprocess.py", line 54, in __init__
    subprocess_orig.Popen.__init__(self, args, 0, *argss, **kwds)
  File "/usr/lib/python2.7/subprocess.py", line 710, in __init__
    errread, errwrite)
  File "/usr/lib/python2.7/subprocess.py", line 1327, in _execute_child
    raise child_exception
"
vagrant@st2-node:~$ 
```

**Comment by userlocalhost** [comment]
And the character encoding that I specified is UTF-8.
```
vagrant@st2-node:~$ echo ｈｏｇｅ | nkf -g
UTF-8
vagrant@st2-node:~$ 
```

**Comment by humblearner** [comment]
Interesting: 

Its works fine on centos72: 

```
[vagrant@centos72 ~]$ echo ｈｏｇｅ | nkf -g
UTF-8
[vagrant@centos72 ~]$ st2 run core.local cmd='echo ｈｏｇｅ'
.
id: 584f7e474c3a5307cfa4e736
status: succeeded
parameters:
  cmd: "echo ｈｏｇｅ"
result:
  failed: false
  return_code: 0
  stderr: ''
  stdout: "ｈｏｇｅ"
  succeeded: true
[vagrant@centos72 ~]$
```

but fails in ubuntu:

```
vagrant@u14:~$ st2  run core.local cmd='echo ｈｏｇｅ'
.
id: 584f8c156d90e8324e46a0d4
status: failed
parameters:
  cmd: "echo ｈｏｇｅ"
result:
  error: '''ascii'' codec can''t encode characters in position 39-42: ordinal not in range(128)'
  traceback: "  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/st2actions/container/base.py", line 100, in _do_run
    (status, result, context) = runner.run(action_params)
  File "/opt/stackstorm/runners/local_runner/local_runner.py", line 152, in run
    kill_func=kill_process)
  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/st2common/util/green/shell.py", line 87, in run_command
    env=env, cwd=cwd, shell=shell, preexec_fn=preexec_func)
  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/eventlet/green/subprocess.py", line 54, in __init__
    subprocess_orig.Popen.__init__(self, args, 0, *argss, **kwds)
  File "/usr/lib/python2.7/subprocess.py", line 710, in __init__
    errread, errwrite)
  File "/usr/lib/python2.7/subprocess.py", line 1327, in _execute_child
    raise child_exception
"
vagrant@u14:~$
```

**Comment by Kami** [comment]
Thanks for the report.

It could be something with the default locale used for new spawned shells. It would be good to see the output of `locale` command (`st2 run core.local cmd=locale` since we are interested in the locale used by python runner sub-process) to see what is going on.

**Comment by userlocalhost** [comment]
Thank you for your reply.

I also confirmed that this problem is invalid in CentOS(v7.2).
```
[vagrant@cent7 etc]$ cat /etc/redhat-release 
CentOS Linux release 7.2.1511 (Core) 
[vagrant@cent7 etc]$ st2 run core.local cmd='echo ｈｏｇｅ'
.
id: 5850bbed2925c03f308151fe
status: succeeded
parameters: 
  cmd: "echo ｈｏｇｅ"
result: 
  failed: false
  return_code: 0
  stderr: ''
  stdout: "ｈｏｇｅ"
  succeeded: true
[vagrant@cent7 etc]$ 
```

> It could be something with the default locale used for new spawned shells.

I see!
Here is the output of `locale` command that st2actionrunner invokes.
```
vagrant@st2-node:~$ cat /etc/lsb-release 
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=14.04
DISTRIB_CODENAME=trusty
DISTRIB_DESCRIPTION="Ubuntu 14.04.5 LTS"
vagrant@st2-node:~$ st2 run core.local cmd='locale'
.
id: 5850bb686f086f4287f46a44
status: succeeded
parameters: 
  cmd: locale
result: 
  failed: false
  return_code: 0
  stderr: ''
  stdout: 'LANG=

    LANGUAGE=

    LC_CTYPE="POSIX"

    LC_NUMERIC="POSIX"

    LC_TIME="POSIX"

    LC_COLLATE="POSIX"

    LC_MONETARY="POSIX"

    LC_MESSAGES="POSIX"

    LC_PAPER="POSIX"

    LC_NAME="POSIX"

    LC_ADDRESS="POSIX"

    LC_TELEPHONE="POSIX"

    LC_MEASUREMENT="POSIX"

    LC_IDENTIFICATION="POSIX"

    LC_ALL='
  succeeded: true
vagrant@st2-node:~$ locale
LANG=en_US.UTF-8
LANGUAGE=
LC_CTYPE="en_US.UTF-8"
LC_NUMERIC="en_US.UTF-8"
LC_TIME="en_US.UTF-8"
LC_COLLATE="en_US.UTF-8"
LC_MONETARY="en_US.UTF-8"
LC_MESSAGES="en_US.UTF-8"
LC_PAPER="en_US.UTF-8"
LC_NAME="en_US.UTF-8"
LC_ADDRESS="en_US.UTF-8"
LC_TELEPHONE="en_US.UTF-8"
LC_MEASUREMENT="en_US.UTF-8"
LC_IDENTIFICATION="en_US.UTF-8"
LC_ALL=
vagrant@st2-node:~$ cat /etc/default/locale 
# Created by cloud-init v. 0.7.5 on Wed, 14 Dec 2016 02:49:04 +0000
LANG=en_US.UTF-8
vagrant@st2-node:~$
```

Thank you.

**Comment by Kami** [comment]
Yeah, it looks like that in action runner / new shell case locale is not `en_US.UTF-8` and that's probably what is causing issues.

I'm not sure why it's not using a default locale, maybe something related to upstart service files or similar.

**Comment by userlocalhost** [comment]
I found another problem about unicode.

I could execute an Action of `local-shell-cmd` successfully as is mentioned before.
```
[dooga@localhost ~]$ st2 run core.local cmd='echo ｈｏｇｅ'
.
id: 58510d66e1382331a2d04bf7
status: succeeded
parameters: 
  cmd: "echo ｈｏｇｅ"
result: 
  failed: false
  return_code: 0
  stderr: ''
  stdout: "ｈｏｇｅ"
  succeeded: true
[dooga@localhost ~]$ st2 action get default.test
```

But `local-shell-script` type one couldn't be executed in the same environment.
```
[dooga@localhost ~]$ st2 action get default.test
+-------------+--------------------------+
| Property    | Value                    |
+-------------+--------------------------+
| id          | 5850f582e138233f0387c9b2 |
| uid         | action:default:test      |
| ref         | default.test             |
| pack        | default                  |
| name        | test                     |
| description |                          |
| enabled     | True                     |
| entry_point | test.sh                  |
| runner_type | local-shell-script       |
| parameters  | {                        |
|             |     "param": {           |
|             |         "type": "string" |
|             |     }                    |
|             | }                        |
| notify      |                          |
| tags        |                          |
+-------------+--------------------------+
[dooga@localhost ~]$ st2 run default.test param='ｈｏｇｅ'
.
id: 58510d71e1382331a2d04bfa
status: failed
parameters: 
  param: "ｈｏｇｅ"
result: 
  error: '''ascii'' codec can''t encode characters in position 0-3: ordinal not in range(128)'
  traceback: "  File "/opt/stackstorm/st2/lib/python2.7/site-packages/st2actions/container/base.py", line 99, in _do_run
    (status, result, context) = runner.run(action_params)
  File "/opt/stackstorm/runners/local_runner/local_runner.py", line 116, in run
    args = action.get_full_command_string()
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/st2common/models/system/action.py", line 179, in get_full_command_string
    return self._format_command()
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/st2common/models/system/action.py", line 183, in _format_command
    positional_args=self.positional_args)
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/st2common/models/system/action.py", line 239, in _get_script_arguments
    command_parts.append('%s=%s' % (quote_unix(arg), quote_unix(str(value))))
"
[dooga@localhost ~]$ 
```

I made the patch for this.
https://github.com/StackStorm/st2/pull/3116

Thank you.

**Comment by userlocalhost** [comment]
This issue has already been fixed in v2.1.1. So I close it.
Thank you.
